frontend_gen.py

```diff
@@ -296,22 +369,35 @@
                                 y=component["y"],
                                 height=component["height"],
                                 width=component["width"],
+                                gridHeight=grid["row_height"],
                                 image_route=f"require('{component['image_route']}')['default']"
                                 if "image_route" in component
                                 else "''",
                             )
                         )
+                        import_set.add(
+                            "const Markdown = React.lazy(() => import('../Markdown'))"
+                        )
                     elif re.match(r"^plot.*$", component["type"]):
                         p.write(
                             fullplotly_template.format(
                                 component_name=component_name,
                                 x=component["x"],
                                 y=component["y"],
+                                gridHeight=grid["row_height"],
                                 height=component["height"],
                                 width=component["width"],
                                 route=component["route"],
+                                channelList=(
+                                    f"channelList={{{component['''channels''']}}}"
+                                    if "channels" in component
+                                    else ""
```

Comment:
Perhaps we can probably push this a bit higher since we perform the check repeatedly.

Reply:
Not possible here, this is making sure that the component only gets access to a channel list if a list of channels is specified in the config. This is something that needs to be decided per component.

Reply:
Sounds good. :+1: